src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -173,8 +188,26 @@ public void NotifyCascadingValueChanged(in ParameterViewLifetime lifetime)
                 ? new ParameterView(lifetime, _latestDirectParametersSnapshot.Buffer, 0)
                 : ParameterView.Empty;
             var allParams = directParams.WithCascadingParameters(_cascadingParameters!);
-            var task = Component.SetParametersAsync(allParams);
-            _renderer.AddToPendingTasks(task);
+            SupplyCombinedParameters(allParams);
+        }
+
+        // This should not be called from anywhere except SetDirectParameters or NotifyCascadingValueChanged.
+        // Those two methods know how to correctly combine both cascading and non-cascading parameters to supply
+        // a consistent set to the recipient.
+        private void SupplyCombinedParameters(ParameterView directAndCascadingParameters)
```

Comment:
Is this change related to the error boundaries work or general goodness?

Reply:
It's to bring the "normal params" and "cascaded params" code paths into a common place so they can share the same interactions with error boundaries (the call to `AddToPendingTasks` which supplies an `owningComponentState`, which in turn determines how we route to the correct error boundary).